src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java

```diff
@@ -313,16 +316,25 @@ private IssueListener createIssueListener() {
       ClientInputFile inputFile = issue.getInputFile();
       if (inputFile != null) {
         URI uri = inputFile.getClientObject();
-        issuesPerFileURI.computeIfAbsent(uri, u -> new ArrayList<>()).add(issue);
+        issuesPerIdPerFileURI.computeIfAbsent(uri, u -> new HashMap<>()).put(UUID.randomUUID().toString(), issue);
```

Comment:
Any possibility of performance impact when there are a lot of issues? Or is UUID generation negligible?

Reply:
I expect the performance impact to be negligible compared to the actual analysis time. I did a quick benchmark on a tight loop with Java 11 vs an implementation based on `AtomicLong`, `UUID` is consistently faster up until ~20k items.